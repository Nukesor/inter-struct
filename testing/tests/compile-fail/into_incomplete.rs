/// This should crash, since the generated initializer for IntoStruct will be incomplete.
use inter_struct::InterStruct;

#[derive(InterStruct)]
#[into("crate::IntoStruct")]
pub struct FromStruct {
    pub normal: String,
}

/// A struct with less, but otherwise identical fields.
pub struct IntoStruct {
    pub normal: String,
    pub optional: Option<String>,
}

fn main() {}
