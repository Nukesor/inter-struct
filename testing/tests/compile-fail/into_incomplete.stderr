error[E0063]: missing field `optional` in initializer of `IntoStruct`
 --> tests/compile-fail/into_incomplete.rs:5:8
  |
5 | #[into("crate::IntoStruct")]
  |        ^^^^^^^^^^^^^^^^^^^ missing `optional`
